streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -295,52 +369,79 @@ public void testMaybePunctuate() throws Exception {
                 new ConsumerRecord<>(partition2.topic(), partition2.partition(), 45, 0L, TimestampType.CREATE_TIME, 0L, 0, 0, recordKey, recordValue)
         ));
 
-        assertTrue(task.maybePunctuate());
+        assertTrue(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(5, task.numBuffered());
         assertEquals(1, source1.numReceived);
         assertEquals(0, source2.numReceived);
 
-        assertFalse(task.maybePunctuate());
+        assertFalse(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(4, task.numBuffered());
         assertEquals(1, source1.numReceived);
         assertEquals(1, source2.numReceived);
 
-        assertTrue(task.maybePunctuate());
+        processorStreamTime.supplier.scheduleCancellable.cancel();
+
+        assertFalse(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(3, task.numBuffered());
         assertEquals(2, source1.numReceived);
         assertEquals(1, source2.numReceived);
 
-        assertFalse(task.maybePunctuate());
+        assertFalse(task.maybePunctuateStreamTime());
```

Comment:
Removed most assertions in this test. Perhaps can remove some more of the remaining ones.